pkg/KubeArmorOperator/internal/controller/resources.go

```diff
@@ -127,6 +127,12 @@ func genRuntimeVolumes(runtime, runtimeSocket, runtimeStorage string) (vol []cor
 					},
 				},
 			})
+
+			// Disguise k0s containerd-storage as "/run/containerd"
+			if storageLocation == "/run/k0s/containerd" {
```

Comment:
Can you make the logic more generic? We should use the disguise method for all runtimes. 

The condition should something be like

Any containerd volume should be mounted to `/run/containerd`

Any docker/crio should be mounted in a similar way to a generic docker/crio path recognised in KubeArmor.

Rest of the changes LGTM. 

@nyrahul we would need a new workflow for setting up k0s and running smoke tests there. Do we want to do this for other runtimes as well like minikube, kind, microk8s?

Reply:
Yes it would be great. However, it would be better to not spawn another VM for minikube based test. The least we can do is trigger these tests only during version release.

Reply:
We probably can't test KinD on GHA (because no BPFLSM), Minikube can be tested in GHA because it uses it's own ISO which support BPFLSM, no issues with rest of them as well.

I will create a separate issue for that.

Reply:
#1417